include/mbgl/style/source.hpp

```diff
@@ -66,12 +66,14 @@ class Source {
     std::string getID() const;
     optional<std::string> getAttribution() const;
 
+    // The data from the volatile sources are not stored in a persistent storage.
```

Comment:
@pozdnyakov thanks for the quick turnaround on this. Implementation looks great. Do we know how likely we are to introduce additional storage options for sources in the future?

From an API extensibility perspective, I'm wondering if it would be better to introduce smth like `Source::getStoragePolicy` and `Source::setStoragePolicy`. If we were to introduce new storage types alongside `volatile` and `permanent` in the future, we wouldn't need to create new APIs and the boolean flags wouldn't need to be deprecated.

cc @tmpsantos @alexshalamov @LukasPaczos @tobrun @1ec5 @julianrex

Reply:
@chloekraw the Carbon API for sources uses [generic parameters](https://github.com/mapbox/mapbox-gl-native-internal/blob/internal/internal/bindings/mbg/map.mbg#L1130), so changing from a boolean flag to a enumeration type in future shall be painless :slightly_smiling_face: